### ISSUE TITLE
Infixes and postfixes should be considered for LTM together

### ISSUE THREAD

**Issue by masak** [masak/007#454]
Back in the '00s, Perl 6 took the natural step to require a distinction between infixes and postfixes:

* If there's whitespace between the term and the operator, it's an infix.
* If there's no whitespace, it's an infix or a postfix.

Meanwhile, in the future, 007 took the "road not taken" by Perl 6. (One of the _few_ remaining opinionated differences between the languages.) In 007, whitespace between "tokens" does not matter anywhere, which means that the above distinction can not exist, which _means_...

```sh
$ bin/007 -e='func infix:<!!>(a, b) { a ~ " !! " ~ b }; func postfix:<!>(s) { s ~ "!" }; say("OH" !!)'
OH!!
```

That shouldn't even parse (because it should prefer the infix due to LTM), and...

```sh
$ bin/007 -e='func infix:<!>(a, b) { a ~ " !! " ~ b }; func postfix:<!!>(s) { s ~ "!" }; say("OH"!!!"HAI")'
OH! !! HAI
```

That shouldn't even parse either, since it should again prefer the infix, and then get stuck at the remaining `!` which could only be a prefix but there isn't one.

Basically, postfixes and infixes need to be thrown into the same pot and LTM'd together. That's the only sane thing to do. Once we get something back we sort out whether it was an infix or a postfix.

In a way, this issue is an echo both of #353 and #409.

I predict that we'll need to leave the cutesy model of `<termish> %% <infix>` behind for the right behavior to emerge. And &mdash; I didn't think I would say this, but here we are &mdash; good riddance. It's more and more come to look like a local optimum that we _should_ get out of. For one, the `not` mentioned in #300 isn't even possible, because listop prefixes aren't possible. Here under this model, they just might be.

I have no more rambling for now.

**Comment by masak** [comment]
I just noticed that this is also currently mis-explained in the user documentation, and would need to be re-worded there also as part of fixing this issue.
